target/arm/translate-a64.c

```diff
@@ -1933,7 +2427,49 @@ static void handle_sys(DisasContext *s, uint32_t insn, bool isread,
 
     tcg_rt = cpu_reg(s, rt);
 
-    if (isread) {
+    ptrdiff_t fieldoffset = ri->fieldoffset;
+
+#ifdef TARGET_CHERI
+    if (ri->restricted_alias_offset && !executive) {
+        fieldoffset = ri->restricted_alias_offset;
+    }
+
+    if (is_morello) {
+        // Existing readfn/writefn functions only apply to the cursor,
+        // it should be ensured they do not cause out of bounds.
+        if (isread) {
+            if (ri->type & ARM_CP_CONST) {
+                assert(0 && "TODO");
```

Comment:
Both?

Reply:
As in, is it a property of the architecture that no such registers exist? If so the assertion should state that, otherwise it looks like we're missing something (and doesn't hint that if this assertion is hit something else has gone wrong rather than this line).